lib/posix/CMakeLists.txt

```diff
@@ -1,17 +1,20 @@
 
 add_library(PTHREAD INTERFACE)
 
+if(CONFIG_POSIX_HEADERS)
 target_include_directories(PTHREAD INTERFACE ${ZEPHYR_BASE}/include/posix)
```

Comment:
this should not be here, or you will not be able to select host files instead of Zephyr posix files and viceversa

Reply:
> this should not be here, or you will not be able to select host files instead of Zephyr posix files and viceversa

???

Assuming you put comment on the exact line, it simply **is** there. Saying that it should not be there means digging in history, and addressing that comment to the author of that line. (But from my PoV, it of course should be there. But of course, it should be controlled, like my patch does.)

Reply:
I know it is there.. ;). Werent you trying to improve it? :p
Done like that you have the problem with inclusion order dependencies (which there is today) in the posix API files, where some of its headers and the host headers have the same name.

Reply:
> I know it is there.. ;). Werent you trying to improve it? :p

So, let's start from the start. When someone wants to use a POSIX function, they go to a reference and look it up. E.g. for sleep() http://pubs.opengroup.org/onlinepubs/9699919799/functions/sleep.html , it says `#include <unistd.h>`. So, using `#include <unistd.h>` for sleep() should just work, ditto for all other POSIX functions. That's the baseline requires for POSIX subsystem (and POSIX subsystem, Zephyr's included).

Being able to use `#include <posix/unistd.h>` is convenience hack, until the above works perfectly. What my patch does is to allow to control of automatic include path for include/posix/ in fine-grained way (as required to to make it work for hundreds of POSIX symbols, one by one).